### ISSUE TITLE
Make addon installations less hostile to dynamic namespaces

### ISSUE THREAD

**Issue by clhodapp** [GoogleCloudPlatform/k8s-config-connector#241]
Older-style Workload Identity installations of Config Connector use a single shared Kubernetes service account to authenticate all operations against the GCP API, regardless of which namespace the Config Connector resources are defined in. Newer-style addon-based Workload Identity installations of Config Connector use separated-out Kubernetes service accounts for each monitored namespace. Overall, this change is great for allowing a more fine-grained security model when the set of namespaces being managed is relatively static (like production environments), but it is a major backslide for systems where namespaces are being brought up and down on a regular basis because it requires extra out-of-band operations to create/remove Workload Identity IAM bindings. My concrete example would be a CI system that uses Kubernetes namespaces to establish dynamic environments for validating each change. Ideally, such a system would be able to perform all of its work against the Kubernetes API and would not need to directly own the management of GCP resources, since delegating that ownership is exactly the point of Config Connector.

It would be wonderful if the `ConfigConnector/configconnector.core.cnrm.cloud.google.com` resource could be augmented with an extra field to select between namespaced and global Workload Identity security in order to retain functional parity with the "manual" installation method.

**Comment by jcanseco** [comment]
Hi @clhodapp, this is actually something we're actively working on at the moment! :) 

More specifically, we're working to add support for "cluster-mode" for the KCC add-on, which will allow you to use one Google Service Account for all the namespaces in your cluster, similar to how traditional "manual" Workload Identity installations worked.

It should be coming as part of an upcoming release.

**Comment by jcanseco** [comment]
Hi @clhodapp, we just released v1.15.0 of the KCC operator which adds support for the cluster-mode set-up I described above. 

Please note that it'll be another few weeks for the KCC add-on to pick up this version of the KCC operator, so you'll have to download and install the KCC operator as a standalone for now if you want to try out cluster-mode today.

**Comment by clhodapp** [comment]
@jcanseco At this moment, it isn't urgent enough to try something short-term, since the manual workload identity installation method still exists. So that I can understand what to expect in the future, though, I will ask: Do new releases out through the GKE channels automatically include the at-the-time-latest version of the Config Connector add-on?

**Comment by jcanseco** [comment]
> Do new releases out through the GKE channels automatically include the at-the-time-latest version of the Config Connector operator?

Yep, that's exactly right. New GKE versions essentially pick up the latest operator version available at the time of its build phase.

**Comment by clhodapp** [comment]
@jcanseco Got it. Thank you for the very quick replies and for the great work that the Config Connector team have put in!

**Comment by jcanseco** [comment]
No problem! Glad to help :)
